src/main/java/com/github/reducktion/socrates/validator/GermanyIdValidator.java

```diff
@@ -0,0 +1,125 @@
+package com.github.reducktion.socrates.validator;
+
+import static com.github.reducktion.socrates.internal.StringUtils.isNumeric;
+import static java.util.stream.Collectors.counting;
+import static java.util.stream.Collectors.groupingBy;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+
+/**
+ * A national identification number does not really exists within Germany.
+ * But there's a tax ident number, which is unique for each citizen and assigned by date of birth.
+ * This one is restricted by law to be used as a marker/attribute for tax/finance relevant purposes only,
+ * and not to identify people in general.
+ * It's called 'steuerliche Identifikationsnummer' or abbreviated 'Steuer-IdNr.'.
+ * There's no meta data, which could be extracted from this number.
+ *
+ * See https://de.wikipedia.org/wiki/Steuerliche_Identifikationsnummer
+ * The algorithm is published in an official document
+ * from: Informations Technik Zentrum Bund
+ * title: Steueridentifikationsnummer (IdNr) nach § 139b AO; Informationen zur Berechnung gültiger Prüfziffern
+ */
+public class GermanyIdValidator implements IdValidator {
+
+    private static final int ID_NUMBER_OF_CHARACTERS = 11;
+    private static final int CHECKSUM_INDEX = 10;
+
+    @Override
+    public boolean validate(final String value) {
+        if (value == null) {
+            return false;
+        }
+        final String sanitizedValue = value.trim();
+        if (sanitizedValue.length() != ID_NUMBER_OF_CHARACTERS
+            || !isNumeric(sanitizedValue)
```

Comment:
Avoid using so much static imports. It might be hard to know where the method comes from.
More info about this [here](https://docs.oracle.com/javase/8/docs/technotes/guides/language/static-import.html).

Reply:
I'm counting 3 static imports. That's not "so much" IMHO.
Other classes in the project have 3 static imports as well.
Also, I see no benefit in replacing line 36 with ```StringUtils.isNumeric(sanitized)...```
I do propose to leave it as is.

Reply:
Fair for you @afbernardino ?

Reply:
Yes, 3 static imports is not "so much" I agree. That was a poor choice of words from my part and I am sorry for that. I didn't want to point out the counting but the usage of it.

The advantage of having `StringUtils.isNumeric(sanitizedValue)` is that it is explicit where the method comes from. The methods from `Collectors` class are easier to grasp, because they are normally used in the streams and developers are used to it.

But, I totally get your point and I think it is fair enough. Therefore, I will resolve the conversation. 👍